torch/testing/_creation.py

```diff
@@ -0,0 +1,159 @@
+"""
+This package contains tensor creation utilities.
+"""
+
+import torch
+from typing import Optional, Tuple, Union
+import math
+
+__all__ = [
+    "make_tensor"
+]
+
+def make_tensor(
+    size: Tuple[int, ...],
+    *,
+    device: Union[str, torch.device] = "cpu",
+    dtype: torch.dtype = torch.float32,
+    low: Optional[float] = None,
+    high: Optional[float] = None,
+    requires_grad: bool = False,
+    noncontiguous: bool = False,
+    exclude_zero: bool = False
+) -> torch.Tensor:
+    r"""Creates a random tensor with the given :attr:`size`.
+
+    The function comes with other optional arguments to allow random tensor generation for the user's needs.
+
+    By default, :attr:`device` is ``cpu`` and :attr:`dtype` is ``torch.float32``, but user can pass these arguments
+    as required.
+
+    If :attr:`low` and :attr:`high` are not passed, following default values are considered depending on the given
+    :attr:`dtype`:
+
+    - boolean type: `low` = 0, `high` = 2.
```

Comment:
1. To indicate inline code, you need double backticks in RST.
2. I would move the default values inside the inline code block
3. The docstring should explain above that the interval is `[low, high)` otherwise `high=2` makes no sense for booleans

```suggestion
    - boolean type: ``low=0``, ``high=2``.
```

Reply:
assert_close also has a cool table explaining the dtype specific defaults, but the defaults for make_tensor are much simpler, so they can probably just be described inline in the description of the low and high args. Whichever arg is described first can contain the full description of how low and high work, and then the arg described second can just say "sets the low/high value of the range, as specified above" to avoid redundancy